pkgs/development/libraries/mesa/default.nix

```diff
@@ -115,7 +110,8 @@ stdenv.mkDerivation {
     "-Dva-libs-path=${placeholder "drivers"}/lib/dri"
     "-Dd3d-drivers-path=${placeholder "drivers"}/lib/d3d"
     "-Dgallium-nine=${boolToString enableGalliumNine}" # Direct3D in Wine
-    "-Dosmesa=${if enableOSMesa then "gallium" else "none"}" # used by wine
+    "-Dosmesa=${boolToString enableOSMesa}" # used by wine
+    "-Dmicrosoft-clc=disabled" # TODO: Auto-feature that requires cmake and llvmPackages.clang
```

Comment:
I'll have a closer look regarding `microsoft-clc` later. Simply adding cmake and llvmPackages.clang is apparently not enough but I'm not sure if we should enable it anyway. Opinions welcome ;)

Reply:
Huh... https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/7565. Seems to be an alternative OpenCL implementation :thinking: ?

Reply:
Seems like it implements the OpenCL 1.2 API on top of D3D12: https://github.com/microsoft/OpenCLOn12 and https://www.collabora.com/news-and-blog/news-and-events/introducing-opencl-and-opengl-on-directx.html.

So I guess right now it is only relevant for Windows (/ DirectX) and we can safely disable it (Nix on Windows users could still enable it later if the need arises). Maybe in the future the OpenCL -> NIR part can be reused for Linux (i.e. offering an alternative to the NIR (new internal representation for the Mesa shader compiler) -> DXIL (DirectX Intermediate Language) step).

Reply:
I wonder if it could be useful for `x86_64-linux` inside WSL, but in any case nothing I'd see as a priority.